sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +691,69 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+//TODO: Timed counterpart for timed AssetLockListener for reversion
+const assetLockCallbackTimed = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    //Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+
+    const reclaimCallback = async function(contract: Contract, assetType: string, assetID: string, recipientECert: string, contractID: string) {
+        //TODO: Call isAssetLockedInHTLC -> Hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract,contractID));
+        
+        if(islocked == false) {
+            //CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            //CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                //throw new Error(`GetHTLCHashPreImageByContractId evaluateTransaction Error: ${evaluateError}`);
+                //If it succeeds, no change
+                const [isReclaimable, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLC(contract, assetType, assetID, recipientECert));
```

Comment:
This is correct, just rename `isReclaimable` to `reclaimResult`, as this function is not checking if asset is reclaimable, instead it is actually reclaiming/unlocking.
Also optionally you can use `reclaimAssetInHTLCusingContractId` there you would just need to pass `contract` and `contractId` as parameter, and you can get rid of other parameters then